src/robenv/ros/ros.py

```diff
@@ -0,0 +1,109 @@
+#
+#  Copyright (c) Honda Research Institute Europe GmbH
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#  1. Redistributions of source code must retain the above copyright notice,
+#     this list of conditions and the following disclaimer.
+#
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+#  3. Neither the name of the copyright holder nor the names of its
+#     contributors may be used to endorse or promote products derived from
+#     this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+#
+
+from __future__ import annotations
+
+import os
+
+from logging import getLogger
+from pathlib import Path
+from shutil import copy
+
+import requests
+
+from robenv.environment.distro import parse_distro
+from robenv.environment.run_command import run_command
+
+
+_logger = getLogger(__name__)
+
+
+class ROS:
+    def __init__(self, path_or_url: str) -> None:
+        cache_path = Path.home() / ".cache/robenv"
+        xdg_home = os.environ.get("XDG_CACHE_HOME")
+
+        self._archive_path = Path()
+        self._distro_path = Path()
+
+        if xdg_home is not None:
+            cache_path = Path(xdg_home) / "robenv"
+
+        if path_or_url.startswith("http") or path_or_url.endswith("tar.gz"):
```

Comment:
ros2 packages are apparently `bz2` packages which wouldn't get caught by this if one would download it and do it from localhost. 

If we go with URL parsing (see other comment) we could just ask for the `scheme`: [urllib.parse.urlparse](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse)

Reply:
i dont see any inprovement with urlparse or do i miss something?